UNIFAC compile errors, closes #1426

JonWel opened the pull request:
This should solve it.

ibell commented on src/Backends/Cubics/UNIFAC.cpp:
```diff
@@ -195,7 +195,9 @@ double UNIFAC::UNIFACMixture::ln_gamma_R(const double tau, std::size_t i, std::s
 }
 void UNIFAC::UNIFACMixture::activity_coefficients(double tau, const std::vector<double> &z, std::vector<double> &gamma){
-    std::size_t N = z.size();
+    if (this->N != z.size()) {
```
Does N still get set somewhere?

JonWel replied:
Yes, [here](https://github.com/CoolProp/CoolProp/blob/master/src/Backends/Cubics/UNIFAC.cpp#L247).
The one I removed is because I do not want it to interfere with `set_components`, which is the place it should be set (ie. I corrected something I forgot to change earlier).

ibell commented on src/Backends/Cubics/UNIFAC.h:
```diff
@@ -20,7 +20,7 @@ namespace UNIFAC
         CoolProp::CachedElement _T; ///< The cached temperature
 
-        std::size_t N = 0; ///< Number of components
+        std::size_t N; ///< Number of components
```
If you did want to initialize this variable, the right place for that is the initializer for the class.

JonWel replied:
`set_components` should always be called, so there is no need to initialise before.
I don't know why I did initialised it to 0.

However, If we plan to use `add_component` alone somewhere to add components after having set them (add_component is public), this may be required to add something in `add_component` that would adapt `N`.

JonWel commented:
Yes, [here](https://github.com/CoolProp/CoolProp/blob/master/src/Backends/Cubics/UNIFAC.cpp#L247).
The one I removed is because I do not want it to interfere with `set_components`, which is the place it should be set (ie. I corrected something I forgot to change earlier).
